internal/gatewayapi/filters.go

```diff
@@ -625,28 +626,18 @@ func (t *Translator) processResponseHeaderModifierFilter(
 		for _, addHeader := range headersToAdd {
 			emptyFilterConfig = false
 			if addHeader.Name == "" {
-				routeStatus := GetRouteStatus(filterContext.Route)
-				status.SetRouteStatusCondition(routeStatus,
-					filterContext.ParentRef.routeParentStatusIdx,
-					filterContext.Route.GetGeneration(),
-					gwapiv1.RouteConditionAccepted,
-					metav1.ConditionFalse,
-					gwapiv1.RouteReasonUnsupportedValue,
+				updateRouteStatusForRHMFilter(
+					filterContext,
 					"ResponseHeaderModifier Filter cannot add a header with an empty name",
 				)
 				// try to process the rest of the headers and produce a valid config.
 				continue
 			}
-			// Per Gateway API specification on HTTPHeaderName, : and / are invalid characters in header names
-			if strings.Contains(string(addHeader.Name), "/") || strings.Contains(string(addHeader.Name), ":") {
-				routeStatus := GetRouteStatus(filterContext.Route)
-				status.SetRouteStatusCondition(routeStatus,
-					filterContext.ParentRef.routeParentStatusIdx,
-					filterContext.Route.GetGeneration(),
-					gwapiv1.RouteConditionAccepted,
-					metav1.ConditionFalse,
-					gwapiv1.RouteReasonUnsupportedValue,
-					fmt.Sprintf("ResponseHeaderModifier Filter cannot set headers with a '/' or ':' character in them. Header: %q", string(addHeader.Name)),
+			if !isModifiableHeader(string(addHeader.Name)) {
+				updateRouteStatusForRHMFilter(
+					filterContext,
+					fmt.Sprintf("ResponseHeaderModifier Filter cannot set headers with a '/' or ':' character in them, or the host header. Header: %q",
```

Comment:
```suggestion
					fmt.Sprintf("Header: %q. The ResponseHeaderModifier filter cannot set the Host header or headers with a '/' or ':' character in them. To modify the Host header use the URLRewrite or the HTTPRouteFilter filter.",
```

Reply:
can we use this in other places too ?

Reply:
Do we want to add this?  The host header should not be modified in the response or be deleted. @arkodg

Reply:
ah this is response, yeah you can update that line, and use this line for requestHeaderModifier, my bad